Source/UI/NewUI/CommonUI/View/Core/TextFieldView.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+protocol TextFieldViewDelegate: AnyObject {
+    func textFieldShouldBeginEditing(textField: UITextField)
+    func textFieldShouldReturn()
+    func textFieldShouldEndEditing(textField: UITextField, replacementString: String)
+    func textFieldShouldChangeCharactersIn(textField: UITextField, replacementString string: String)
+}
+
+class TextFieldView: UIView {
+    weak var delegate: TextFieldViewDelegate?
+
+    private(set) lazy var textField: UITextField? = {
```

Comment:
It would be better to completely protect UI components internally. Being reference types, the modifications happening elsewhere have consequences within the object.

Reply:
please suggest code